sympy/core/tests/test_symbol.py

```diff
@@ -366,3 +367,24 @@ def test__uniquely_named_symbol_and__symbol():
     assert F('x1', Symbol('x1'),
         modify=lambda i: i + '_').name == 'x1_'
     assert _symbol(x, _x) == x
+
+
+def test_disambiguate():
+    x, y, _x, _x_1, _y, _y_1 = symbols('x y _x _x_1 _y _y_1')
+    t1 = _y, Dummy('y'), _x, Dummy('x'), Dummy('x')
+    t2 = Dummy('x'), Dummy('x')
+    t3 = Dummy('x'), Dummy('y')
+    t4 = x, Dummy('x')
+    t5 = Symbol('x', integer=True), x
```

Comment:
If we add `Symbol('x_1') to `t5` I think we will run into problems: the index of the duplicate symbol cannot be added blindly without checking to see if the proposed symbol name is already in the list of known symbols. This might also be a reason to strip underscores *and* numbers (like `'_x_2_3_'.strip('_1234567890') -> x`).